techniques/css/C40.html

```diff
@@ -2,148 +2,109 @@
-               color - then it will always have sufficient contrast against any background color.
-               Currently, this can be done by combining the <code>text-shadow</code> property with the <code>outline</code> property on the <code>focus</code> indicator.
-            </p>
-            <p>Developers may apply this technique to parts of the site where its hard to keep track
-               of the focus indicator (such as when there are lots of different component colors).
-               Developers can also provide a single-color focus indicator for components such as
-               a navigation menus that are used across a site, have specific design requirements,
-               and are easy to test and maintain.
-            </p>
+            <h2>解説</h2>
+            <p>この達成方法は、どのような背景色に対しても十分なコントラストをもつ2色で構成されたフォーカスインジケータを作成することを目的とする。この達成方法により、異なる背景色でフォーカスインジケータを表示したときに十分なコントラストを確保するために複数のクラスを付与する必要なくなる。</p>
+            <p>ブラウザが提供する標準のフォーカスインジケータは多くの場合は単色であるため、フォーカスの視認性は、あるコンポーネントでは高い一方、他のコンポーネントでは低くなる。例えば、フォーカスインジケータが濃い青色で、ボタンが白色の背景上にある場合は見やすいが、背景も青色の場合は見にくい。</p>
+            <p>サイト内の様々なボタンに対応するため、それぞれ固有の CSS クラスを作成することは可能であるが、時間がかかる上、インタラクティブなコンテンツの種類を見逃してしまう可能性がある。そこで、フォーカスインジケータが明るい色と暗い色の2色から構成されていれば、どのような背景色に対しても十分なコントラストを確保することができる。現在、<code>focus</code> インジケータに対して <code>text-shadow</code> プロパティと <code>outline</code> プロパティを組み合わせて使うことで実現できる。</p>
+            <p>開発者は、（異なる色のコンポーネントが多数存在する等の理由で）サイトのフォーカスインジケータを把握することが困難な部分に対して、この方法を適用することができる。また、ナビゲーションメニューのようなサイト全体で使用され、明確なデザイン要件があり、テストやメンテナンスが容易なコンポーネントについては、単色のフォーカスインジケータを提供することができる。</p>
             <div class="note">
-               <div role="heading" class="note-title marker" aria-level="3">Note</div>
-               <p>If it can be determined that the two-color focus indicator CSS takes precedence then
-                  the test can be applied programmatically rather than by manually focusing on each
-                  interface component.
-               </p>
+               <div role="heading" class="note-title marker" aria-level="3">注記</div>
+               <p>2色フォーカスインジケータの CSS が優先されると断定可能な場合は、検証は各インターフェースコンポーネントを手動でフォーカスする代わりに機械的に行うことができる。</p>
             </div>
          </section>
          <section id="examples">
-            <h2>Examples</h2>
-            <p>The examples demonstrate a simple implementation where focus styles are applied to
-               all links and inputs. In use on a more complex website care would need to be taken
-               that these styles are not overridden by more specific styles.
-            </p>
+            <h2>関連リソース</h2>
+            <p>これらの例では、フォーカススタイルがすべてのリンクと入力に適用される、シンプルな実装例を示す。より複雑な Web サイトで使用する場合は、これらのスタイルがより具体的なスタイルによって上書きされないよう、注意が必要である。</p>
             <section class="example" id="example-1-a-light-and-dark-dotted-indicator">
-               <h3>Example 1: A light and dark dotted indicator</h3>
-               <p>Description</p><pre><code>*:focus {
+               <h3>事例 1: 明暗のあるドット状のインジケータ</h3>
```

Comment:
https://eow.alc.co.jp/search?q=dotted

```suggestion
               <h3>事例 1: 明暗のある点線のインジケータ</h3>
```